examples/chip-tool-darwin/BUILD.gn

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2020 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/build.gni")
+import("//build_overrides/chip.gni")
+
+import("${chip_root}/build/chip/tools.gni")
+
+assert(chip_build_tools)
+
+config("includes") {
+  include_dirs = [
+    ".",
+    "${chip_root}/src/darwin/Framework",
+    "${chip_root}/src/darwin/Framework/CHIP",
+  ]
+}
+
+executable("chip-tool-darwin") {
+  sources = [
+    "${chip_root}/src/app/clusters/media-playback-client/media-playback-client.cpp",
```

Comment:
We ought to dedup explicitly listing all the bits in src/app/util using source_set dependencies.

And also for src/darwin/Framework/CHIP

Reply:
Do you have good suggestion for that ? 
My understanding is that I should normally create a `BUILD.gn` file into `src/app/` or `src/app/util` with something like `source_set("chip-app-util") { ...}` into it,  and then add `public_deps += ["${chip_root}/src/app/util:chip-app-util"]` to `examples/*/BUILD.gn` files.

But this naive approach won't work because some of the files from `src/app` depends on `examples/*/gen/` folders :/

Reply:
Ah I see. There are still some options to dedup, such as making a template and invoking it from each application build file, but no silver bullet.